### PR TITLE
[lldb] Support changes to TLS on macOS

### PR DESCRIPTION
**Pull request by bulbazord**
I decided to resubmit #77854 but broken up into multiple commits for better reviewability. I recommend reviewing each commit individually, the overall change is quite large.

rdar://120676969

### CODE REVIEW

**Review by JDevlieghere — APPROVED** [review]
LGTM. Thanks for breaking this up!

**Review by felipepiovezan — APPROVED** [review]
LGTM!

File: lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderDarwin.cpp [hunk 1]
```diff
@@ -1048,74 +1048,103 @@ DynamicLoaderDarwin::GetThreadLocalData(const lldb::ModuleSP module_sp,
...
-        }
+  DataExtractor data(buf, sizeof(buf), m_process->GetByteOrder(), addr_size);
+  lldb::offset_t offset = 0;
+  const lldb::addr_t tls_thunk = data.GetAddress(&offset);
+  const lldb::addr_t key = data.GetAddress(&offset);
+  const lldb::addr_t tls_offset = data.GetAddress(&offset);
+
+  if (tls_thunk != 0) {
+    Address thunk_load_addr;
+    if (target.ResolveLoadAddress(tls_thunk, thunk_load_addr)) {
```
Comment (jasonmolenda) [inline lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderDarwin.cpp:1119]: I'm not sure I understand this sequence.  `tls_thunk` is a virtual address of a function.  We create an Address object with it, then get the load address from the Address object and pass that to our `evaluate_tls_address`.  Why did we roundtrip through an Address?  Did I misunderstand and we've got a file address somewhere?

Comment (bulbazord) [reply lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderDarwin.cpp:1119]: `ThreadPlanCallFunction` takes an `Address`. No file address involved. Is there a simpler way to do this?

Comment (jasonmolenda) [reply lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderDarwin.cpp:1119]: Ah sorry I missed that it was used in the ThreadPlanCallFunction.  I saw `addr_t tls_load_addr = tls_addr.GetLoadAddress()` which is just getting the load address out of the Address object and didn't see we were still using hte Address object.  

Minor nit, but I don't like to prefix `addr_t`s with the `lldb::` namespace, except in headers where it's required.

Comment (bulbazord) [reply lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderDarwin.cpp]: > Minor nit, but I don't like to prefix addr_ts with the lldb:: namespace, except in headers where it's required.
I'm also not a fan... I was mostly preserving existing convention here. I don't mind removing them :)

**Review by jasonmolenda — APPROVED** [review]

**Pull request merged**